pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaPartitionLevelConsumer.java

```diff
@@ -57,6 +59,14 @@ public MessageBatch<byte[]> fetchMessages(long startOffset, long endOffset, int
           endOffset, timeoutMillis);
     }
     _consumer.seek(_topicPartition, startOffset);
```

Comment:
I guess a better way would be to just do `_consumer.seek(_topicPartition, Max(startOffset, beginningOffset))`. Will help us avoid 2 seeks in case of failure.

Reply:
definitely, can do that. 
i'm hoping to find a solution which will avoid the `_consumer.beginningOffsets` call